core/src/main/scala/org/apache/spark/sql/delta/commands/DescribeDeltaDetailsCommand.scala

```diff
@@ -134,11 +120,20 @@ case class DescribeDeltaDetailCommand(
     }
   }
 
-  private def toRows(detail: TableDetail): Seq[Row] = TableDetail.toRow(detail) :: Nil
+  protected def getDescribeDetailSchema : StructType = {
+    ScalaReflection.schemaFor[TableDetail].dataType.asInstanceOf[StructType]
+  }
+
+  private def toRows(sparkSession: SparkSession, detail: TableDetail): Seq[Row] = {
+    import sparkSession.implicits._
+    Seq(detail).toDF().collect().toSeq
```

Comment:
Could you explain why we need to change into this way? This will create a new DataFrame and may run a Spark job (there may be some optimization to save the job) so it will be slower than the current approach.